ANMN/LTSP/TSaggregator/geoserverCatalog_generic.py

```diff
@@ -0,0 +1,112 @@
+    return(vargs)
+
+
+def get_urls(varname=None, site=None, featuretype=None, fileversion=None, realtime=None, timestart=None, timeend=None, outFileList=None):
+
+    """
+    get the urls from the geoserver moorings_all_map collection
+    based on user defined filters
+    """
+
+    if realtime:
+        if realtime.lower() == "yes":
+            url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv&CQL_FILTER=(realtime=TRUE)"
+        elif realtime.lower() == "no":
+            url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv&CQL_FILTER=(realtime=FALSE)"
+        else:
+            sys.exit('ERROR: %s is not yes or no' % realtime)
+    else:
+        url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv"
+
+
+    df = pd.read_csv(url)
+    criteria_all = df.url != None
+
+    ## get possible values
+    separator = ', '
+    varnames_all = set(separator.join(list(df.variables)).split(', '))
+    varnames_all = set(str(list(df.variables)).split(", "))
+    site_all = list(df.site_code.str.upper().unique())
+    featuretype_all = list(df.feature_type. str.lower().unique())
+    fileversion_all = list(df.file_version.unique())
+    mode_all = ['realtime', 'delayed', 'all']
+
+
+    if varname:
+        if varname in varnames_all:
+            criteria_all = criteria_all & df.variables.str.contains('.*\\b'+varname+'\\b.*', regex=True)
+        else:
+            sys.exit('ERROR: %s not in the variable list' % varname)
+
+    if site:
+        if site in site_all:
+            criteria_all = criteria_all & df.site_code.str.contains(site, regex=False)
+        else:
+            sys.exit('ERROR: %s is not in the site_code list' % site)
+
+    if featuretype:
+        if featuretype in featuretype_all:
+            criteria_all = criteria_all & (df.feature_type.str.lower() == featuretype.lower())
+        else:
+            sys.exit('ERROR: %s is not in the feature_type list' % featuretype)
+
+    if fileversion:
+        if fileversion in fileversion_all:
+            criteria_all = criteria_all & (df.file_version == fileversion)
+        else:
+            sys.exit('ERROR: %s is not in the fileversion list' % featuretype)
+
+    if timestart:
+        try:
+            criteria_all = criteria_all & (pd.to_datetime(df.time_coverage_end) >= datetime.strptime(timestart, '%Y-%m-%d'))
+        except ValueError:
+            sys.exit('ERROR: invalid start date.')
+
+    if timeend:
+        try:
+            criteria_all = criteria_all & (pd.to_datetime(df.time_coverage_start) <=  datetime.strptime(timeend, '%Y-%m-%d'))
+        except ValueError:
+            sys.exit('ERROR: invalid end date.')
+
+    if outFileList:
+        df.url[criteria_all].to_csv(outFileList, index=False)
```

Comment:
The `get_urls` function should simply return the selected urls as a list (this will be the input for the aggregator function). The "__main__" script at the bottom can save it to a file, if requested. Otherwise I think the default should be to simply print it to stdout.

Reply:
saving the result in a file was now in  __main__. Argument `-outfileName` removed from the function declaration